chain/chain/src/store.rs

```diff
@@ -2081,6 +2043,144 @@ impl<'a> ChainStoreUpdate<'a> {
+            }
+            DBCol::ColNextBlockWithNewChunk => {
+                store_update.delete(col, key);
+                self.chain_store.next_block_with_new_chunk.cache_remove(key);
+            }
+            DBCol::ColStateHeaders => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColBlock => {
+                store_update.delete(col, key);
+                self.chain_store.blocks.cache_remove(key);
+            }
+            DBCol::ColBlockExtra => {
+                store_update.delete(col, key);
+                self.chain_store.block_extras.cache_remove(key);
+            }
+            DBCol::ColNextBlockHashes => {
+                store_update.delete(col, key);
+                self.chain_store.next_block_hashes.cache_remove(key);
+            }
+            DBCol::ColChallengedBlocks => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColBlocksToCatchup => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColStateChanges => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColBlockRefCount => {
+                store_update.delete(col, key);
+                self.chain_store.block_refcounts.cache_remove(key);
+            }
+            DBCol::ColReceiptIdToShardId => {
+                store_update.delete(col, key);
+                self.chain_store.receipt_id_to_shard_id.cache_remove(key);
+            }
+            DBCol::ColTransactions => {
+                store_update.delete(col, key);
+                self.chain_store.transactions.cache_remove(key);
+            }
+            DBCol::ColChunks => {
+                store_update.delete(col, key);
+                self.chain_store.chunks.cache_remove(key);
+            }
+            DBCol::ColChunkExtra => {
+                store_update.delete(col, key);
+                self.chain_store.chunk_extras.cache_remove(key);
+            }
+            DBCol::ColPartialChunks => {
+                store_update.delete(col, key);
+                self.chain_store.partial_chunks.cache_remove(key);
+            }
+            DBCol::ColInvalidChunks => {
+                store_update.delete(col, key);
+                self.chain_store.invalid_chunks.cache_remove(key);
+            }
+            DBCol::ColChunkHashesByHeight => {
+                store_update.delete(col, key);
+                self.chain_store.chunk_hash_per_height_shard.cache_remove(key);
+            }
+            DBCol::ColStateParts => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColState => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColTrieChanges => {}
+            DBCol::ColBlockPerHeight => {
+                panic!("Must use gc_col_glock_per_height method to gc ColBlockPerHeight");
+            }
+            _ => {}
```

Comment:
@ailisp @Kouprin let's make sure that the columns that require gc are actually gc'ed. So it's better to list all the other variants explicitly here rather than using a wildcard.

Reply:
Agree this is a better idea. For these noop gc columns, do we disallow calling them by panic?

Reply:
Maybe `debug_assert` is better

Reply:
Please choose `unimplemented!` macro as a sign to creators of new columns to add them here. All noop columns should be explicitly added to the list with comment why we don't GC them.

For now, the best option I see is to add noop section to `clear_block_data` as well and call `gc_col` for all rest columns explicitly. It will help us to make sure that columns are truly increase their gc execution number and are not forgotten by mistake.

Reply:
Have a question on no-op columns:
- where their gc_col method should be called if they're no-op? currently, no code call gc_col is called
- should gc_count be increased (and how will no-op columns gc count be checked in StoreValidator)

Reply:
Made all column explicit now. Seems there's no need for unimplemented, since if you're missing a column rust doesn't compile, so as we add new columns we must have add either a gc method, or add it as no-op

Reply:
Can you put all the Columns that are not supposed to be GCed to a Constant Set somewhere is `store.rs`? Then we should panic every time when running GC from the columns from the Set. It will help us to call `gc_col` only from the columns what are not in the Set.

Store Validator will check, that all columns that are not in the Set should increase gc_count values and all columns that are in the Set should remain zeroes.

Reply:
> Store Validator will check, that all columns that are not in the Set should increase gc_count values and all columns that are in the Set should remain zeroes.

I see! Was confused about no-op vs zero gc counts